### PR TITLE
 xbrl and dbf integration of purchased power ferc1 table

### PR DESCRIPTION
**Pull request by cmgosnell**
Task list in issue #1820

This was very straightforward overall. But I added two non-parameterized transform steps into the overwritten version of `transform_main`:
- a bare `drop_duplicates` bc there are 7 complete duplicates from 1998 and 2 from 2022.
- running the encoder from the metadata classes. I could add this into the standard `transform_main` bc is one of the things that does nothing if there is nothing to encode.

Questions:
- should I do any verification with this bare `drop_duplicates`?
- should I move the encoder into the default `transform_main`?
- should I do any wrapping of either or both of these bespoke transforms in the transformer infrastructure? Like make a transformer class/ transform params/ etc?

### CODE REVIEW

File: src/pudl/transform/ferc1.py [hunk 1]
```diff
@@ -849,6 +849,33 @@ class PlantsPumpedStorageFerc1TableTransformer(Ferc1AbstractTableTransformer):
...
+        df = (
+            super()
+            .transform_main(df)
+            # Drop records containing no useful data and also any completely duplicate
+            # records -- there are 6 in 1998 for utility 238 for some reason...
+            .drop_duplicates()
+            .pipe(
+                pudl.metadata.classes.Package.from_resource_ids()
+                .get_resource(self.table_id.value)
+                .encode
```
Comment (zaneselvans) [inline src/pudl/transform/ferc1.py:873]: Oooh, our first encoding on FERC data?

If you're using the encoder on this table shouldn't there also be a new encoding table defined with the acceptable values, and don't the foreign key relationships that cause the encoding need to be defined for the resource?

File: src/pudl/transform/ferc1.py [hunk 2]
```diff
@@ -1361,97 +1388,6 @@ def plant_in_service(ferc1_raw_dfs, ferc1_transformed_dfs):
     return ferc1_transformed_dfs
 
 
-def purchased_power(ferc1_dbf_raw_dfs, ferc1_xbrl_raw_dfs, ferc1_transformed_dfs):
```
Comment (zaneselvans) [inline src/pudl/transform/ferc1.py:1364]: Deleted!!!! 🔥

**Review by zaneselvans — CHANGES_REQUESTED** [review]
Hey, only concern here is what the encoding step is doing, given that there are no foreign key relationships or coding tables defined for the table, are there?

File: src/pudl/transform/ferc1.py (hunk 1 above)
Comment (cmgosnell) [reply src/pudl/transform/ferc1.py:873]: this table was being encoded  in the deleted function... but i was wondering why it was being used at all.

Comment (cmgosnell) [reply src/pudl/transform/ferc1.py:873]: the `purchase_type_code` is the column being encoded

**Review by zaneselvans — APPROVED** [review]

**Pull request merged**